src/hooks/custom-hooks/useOAuth.ts

```diff
@@ -37,18 +38,28 @@ const useOAuth = (): UseOAuthReturn => {
     const authTokenLocalStorage = localStorage.getItem('authToken');
 
     const WSLogoutAndRedirect = async () => {
-        await logout();
-        removeCookies('affiliate_token', 'affiliate_tracking', 'utm_data', 'onfido_token', 'gclid');
-        if (!isOAuth2Enabled) {
-            window.open(oauthUrl, '_self');
+        try {
+            await logout();
+        } catch (error) {
+            // eslint-disable-next-line no-console
+            console.error('Failed to logout', error);
         }
+        removeCookies('affiliate_token', 'affiliate_tracking', 'utm_data', 'onfido_token', 'gclid');
+        redirectToAuth();
     };
+
     const handleLogout = async () => {
-        await OAuth2Logout({
-            postLogoutRedirectUri: window.location.origin,
-            redirectCallbackUri: `${window.location.origin}/callback`,
-            WSLogoutAndRedirect,
-        });
+        try {
+            await OAuth2Logout({
+                postLogoutRedirectUri: window.location.origin,
+                redirectCallbackUri: `${window.location.origin}/callback`,
+                WSLogoutAndRedirect,
+            });
+        } catch (error) {
+            // eslint-disable-next-line no-console
+            console.error('Failed to handle logout', error);
```

Comment:
@amam-deriv are you handling the try catch in Oauth2Logout?
If not we need to do this:
```suggestion
            console.error('Failed to handle logout', error);
            if (err instanceof OIDCError) {
            const currentDomain = window.location.hostname.split('.').slice(-2).join('.');
             Cookies.set('logged_state', 'true', {
                        expires: 30,
                        path: '/',
                        domain: currentDomain,
                        secure: true,
                    });
            }
```

Reply:
As @amam-deriv mentioned, this is not the proper way since `Oauth2Logout` already sets the logged_state to `false` when `post_logout` is completed successfully without errors, so we don't need to manipulate the cookies here